### PR TITLE
[SPARK-17648][CORE] TaskScheduler really needs offers to be an IndexedSeq

### PR DESCRIPTION
**Pull request by squito**
## What changes were proposed in this pull request?

The Seq[WorkerOffer] is accessed by index, so it really should be an
IndexedSeq, otherwise an O(n) operation becomes O(n^2).  In practice
this hasn't been an issue b/c where these offers are generated, the call
to `.toSeq` just happens to create an IndexedSeq anyway.I got bitten by
this in performance tests I was doing, and its better for the types to be
more precise so eg. a change in Scala doesn't destroy performance.
## How was this patch tested?

Unit tests via jenkins.

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala [hunk 1]
```diff
@@ -216,7 +216,7 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
       val activeExecutors = executorDataMap.filterKeys(executorIsAlive)
       val workOffers = activeExecutors.map { case (id, executorData) =>
         new WorkerOffer(id, executorData.executorHost, executorData.freeCores)
-      }.toSeq
+      }.toIndexedSeq
```
Comment (squito) [inline core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala]: this just happened to generate an `IndexedSeq` before in any case, which is why this wasn't a performance bottleneck.  But if scala libs and decided to generate something else on the call to `.toSeq`, scheduling performance would tank before this change.

**Comment by squito** [comment]
Jenkins, retest this please

**Comment by markhamstra** [comment]
I'm glad Jenkins finally got it together. LGTM.

**Comment by andrewor14** [comment]
This looks reasonable. Merging into master. I will leave it out from branch-2.0 just in case.

**Pull request closed without merging**